R/PCA_protti.R

```diff
@@ -0,0 +1,85 @@
+#' Plot principal component analysis
+#'
+#' Plots a principal component analysis based on peptide or precursor intensities.
+#'
+#' @param data A dataframe containing  sample names, peptide or precursor identifiers, corresponding intensities and a condition column indicating e.g. the treatment.
+#' @param sample The column in the data dataframe containing the sample name.
+#' @param grouping The column in the data dataframe containing either precursor or peptide identifiers.
+#' @param intensity Column containing the corresponding intensity values for each peptide or precursor.
+#' @param condition Column indicating the treatment or condition for each sample.
+#'
+#' @return A plotted principal component analysis showing PC1 and PC2
+#' @import dplyr
+#' @import ggplot2
+#' @import tidyr
+#' @import magrittr
+#' @importFrom stringr str_replace
+#' @importFrom stats prcomp
+#' @importFrom rlang .data
+#' @importFrom ggrepel geom_text_repel
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' PCA_protti(
+#' data,
+#' sample = r_file_name,
+#' grouping = pep_stripped_sequence,
+#' intensity = normalised_intensity_log2,
+#' condition = r_condition
+#' )
+#' }
+#'
+PCA_protti <-
+  function(data, sample, grouping, intensity, condition)
+  {
+    . = NULL
+
+    pca_input <- data %>%
+      dplyr::filter(!is.na({{intensity}})) %>%
+      dplyr::select({{sample}}, {{grouping}}, {{intensity}}) %>%
+      dplyr::group_by({{sample}}, {{grouping}}) %>%
+      dplyr::summarise(intensity = sum({{intensity}})) %>%
+      tidyr::pivot_wider(names_from = {{sample}}, values_from = intensity) %>%
+      tidyr::drop_na() %>%
+      dplyr::select(-{{grouping}}) %>%
+      t(.)
+
+    annotation <- data %>%
+      dplyr::select({{sample}}, {{condition}}) %>%
+      magrittr::set_colnames(c("sample", "condition"))
+
+    pca <- prcomp(pca_input, center = TRUE)
+
+    pca_df <- as.data.frame(pca$x) %>%
+      dplyr::mutate(sample = factor(row.names(.))) %>%
+      dplyr::left_join(annotation, by = "sample") %>%
+      dplyr::distinct()
+
+    pca_sdev_df <- as.data.frame(pca$sdev)
+
+    pca_sdev_df <- pca_sdev_df %>%
+      dplyr::mutate(percent_variance = (pca$sdev ^ 2 / sum(pca$sdev ^ 2) * 100),
+             dimension = row.names(.))
+
+    plot <- pca_df %>%
+      ggplot2::ggplot(aes(.data$PC1, .data$PC2, col = condition)) +
```

Comment:
You could also add information about digestion type (lip or tc) as shape. 
Then you would need digestion as an additional argument to the function that is by default NULL so that everything still works even if not provided.